src/pages/VenuePage/VenuePage.tsx

```diff
@@ -38,25 +42,105 @@ import { useVenueId } from "hooks/useVenueId";
 // import { useVenueAccess } from "hooks/useVenueAccess";
 import useConnectCurrentVenue from "hooks/useConnectCurrentVenue";
 
-import { CountDown } from "components/molecules/CountDown";
-import { LoadingPage } from "components/molecules/LoadingPage/LoadingPage";
-// import { AccessDeniedModal } from "components/atoms/AccessDeniedModal/AccessDeniedModal";
-import TemplateWrapper from "./TemplateWrapper";
-
 import { updateTheme } from "./helpers";
 
+import Login from "pages/Account/Login";
+
 import "./VenuePage.scss";
 
-import Login from "pages/Account/Login";
-import { useRelatedVenues } from "hooks/useRelatedVenues";
+const CountDown = React.lazy(() =>
+  import("components/molecules/CountDown").then((m) => ({
+    default: m.CountDown,
+  }))
+);
+const LoadingPage = React.lazy(() =>
+  import("components/molecules/LoadingPage/LoadingPage").then((m) => ({
+    default: m.LoadingPage,
+  }))
+);
+const TemplateWrapper = React.lazy(() => import("./TemplateWrapper"));
 
 // @debt Refactor this constant into settings, or types/templates, or similar?
 const hasPaidEvents = (template: VenueTemplate) => {
   return template === VenueTemplate.jazzbar;
 };
 
+interface PreloadProps {
+  venue?: AnyVenue;
+}
+
+const Preload: React.FC<PreloadProps> = ({ venue }) => (
+  <Helmet>
+    {
+      // preload the icon only when TemplateWrapper with the navbar icon aren't loaded lazily
+      // <link href={SPARKLE_ICON} rel="preload" as="image" />
+    }
+    {venue?.mapBackgroundImageUrl && (
+      <link href={venue?.mapBackgroundImageUrl} rel="preload" as="image" />
+    )}
+    {
+      // NOTE: if there is significant number of rooms, preload might degrade performance, cut down to few
+      (venue?.rooms ?? []).map(
+        (room) =>
+          room?.image_url && (
+            <link
+              key={room?.image_url}
+              href={room.image_url}
+              rel="preload"
+              as="image"
+            />
+          )
+      )
+    }
+  </Helmet>
+);
```

Comment:
Do we have any hard evidence that this extra complexity has much if any real world impact? And even moreso, that would have impact beyond what just generally setting the cache headers appropriately on the map image (and all images loaded from firebase storage)?

How does preload interact with non-cacheable images? Does it have any actual impact, or will it just have to fetch the asset again regardless?

Reply:
Superseded with hook `usePreloadAssets` at e9da1d5788e915f6de13a7aa0315601017c75b73 as per https://github.com/sparkletown/sparkle/pull/1713#discussion_r665881498 which simplifies code a bit.

On the impact side, tested with venues having half a dozen rooms, no browser messages complaining assets got loaded but weren't used for the room images. Using the hook should also not have issues with similar number of rooms.

 Haven't tested on anything like 20, 50, but it's a reasonable assumption that many requests might have impact.

Reply:
If there are no objections, I can resolve this conversation, otherwise, as alternative, could open PR to fix any possible issue rising from large number of rooms i.e. preload assets.